internal/ingress/controller/parser/parser.go

```diff
@@ -497,10 +498,20 @@ func (p *Parser) parseIngressRules(
 func (p *Parser) fillOverrides(state KongState) error {
 	for i := 0; i < len(state.Services); i++ {
 		// Services
+		svc, err := p.store.GetService(
+			state.Services[i].Namespace,
+			state.Services[i].Backend.ServiceName)
+		var anns map[string]string
+		if err != nil {
+			log.Printf("Err getting services %v", err)
+		}
+		anns = svc.Annotations
 		kongIngress, err := p.getKongIngressForService(
-			state.Services[i].Namespace, state.Services[i].Backend.ServiceName)
-		if err == nil {
-			overrideService(&state.Services[i], kongIngress)
+			state.Services[i].Namespace,
+			state.Services[i].Backend.ServiceName)
+
+		if err == nil || kongIngress == nil {
```

Comment:
This logic is a lot more complex than it needs to be.
A simpler solution here would be:
1. call GetService, if err, then log.
2. call GetKongIngressForService, if err then log.
3. Finally, call the override method. If there are errors above, pass in empty/nil structs.

Reply:
To be clear, are you also suggesting stripping out this chunk?
```
		} else {
			glog.Error(errors.Wrapf(err, "fetching KongIngress for service %v/%v",
				state.Services[i].Namespace,
				state.Services[i].Backend.ServiceName))
		}```